src/main/drivers/bus_i2c_stm32f10x.c

```diff
@@ -226,14 +226,14 @@ void i2c_ev_handler(void)
         I2Cx->CR1 &= ~0x0800;                                           // reset the POS bit so ACK/NACK applied to the current byte
         I2C_AcknowledgeConfig(I2Cx, ENABLE);                            // make sure ACK is on
         index = 0;                                                      // reset the index
-        if (reading && (subaddress_sent || 0xFF == reg)) {              // we have sent the subaddr
+        if (reading && (subaddress_sent || -1 == reg)) {                // we have sent the subaddr
```

Comment:
+1
I did't like it, too; I just honored the "Constant-Lefty" style of the original code. Will change it.